src/JWK.js

```diff
@@ -225,6 +295,74 @@ class JWK {
     let { alg, cryptoKey } = this
     return JWA.decrypt(alg, cryptoKey, ciphertext, iv, tag, aad)
   }
+
+  /**
+   * thumbprint
+   *
+   * @description
+   * Calculate the SHA-256 JWK Thumbprint according to [RFC7638]{@link https://tools.ietf.org/html/rfc7638}.
+   * This method is used to create a unique `kid` if none is specified.
+   *
+   * @example <caption>SHA-256 Thumbprint</caption>
+   * jwk.thumbprint()
+   *   .then(console.log)
+   * //
+   * // (line breaks for display only)
+   * //
+   * // => "45BLsBiWcghaEf_NF70Gf5oQcYLHaA
+   * //     tks0C48tT5SJ4"
+   *
+   * @return {Promise.<String>} A promise that resolves the JWK Thumbprint String.
+   */
+  thumbprint () {
+    return JWK.thumbprint(this)
+  }
+
+  /**
+   * generateProtected
+   *
+   * @description
+   * Use key metadata to generate a JWS protected header object.
+   *
+   * @example <caption>Basic JWS Header with JWC</caption>
+   * jwk.generateProtected({ jwc: 'base64url encoded compact jwc' })
+   * // => { alg: 'RS256',
+   * //      kid: 'abcd123$',
+   * //      jwc: 'base64url encoded compact jwc' }
+   *
+   * @example <caption>Basic JWS Header with JKU</caption>
+   * jwk.generateProtected({ jku: 'https://example.com/jwks' })
+   * // => { alg: 'RS256',
+   * //      kid: 'abcd123$',
+   * //      jku: 'https://example.com/jwks' }
+   *
+   * @param  {Object} props - Additional properties to include in header.
+   * @return {Object} JWS Header
+   */
+  generateProtected (props) {
+    let { alg, kid, key_ops, use } = this
+    let header = Object.assign({}, { alg, kid }, props)
```

Comment:
Would it make sense to reverse the order of that Object.assign? Sort of like:

`let header = Object.assign({}, props, { alg, kid })`

That way, properties in `props` won't be able to override the `alg` and `kid`.

Reply:
Just remembered why I did this; its specifically for the case where the key may not have an `alg` or `kid` property and if one was passed in via `props` it would be overwritten with `undefined`.

Reply:
Wonder if  `getProtectedHeader(params)` might be a better method name?

Reply:
A good suggestion. Updated.